torchtune/data/_prompt_templates.py

```diff
@@ -151,13 +156,15 @@ class ChatMLTemplate(PromptTemplateInterface):
     def __call__(
         self,
         messages: List[Message],
+        inference: bool = False,
     ) -> List[Message]:
         """
         Format user, assistant, and system messages with appropriate tags.
 
         Args:
             messages (List[Message]): a single conversation, structured as a list
                 of `Message` objects
+            inference (bool): Whether the template is being used for inference or not.
```

Comment:
probably not needed for this PR, but this is a good opportunity to use the inference flag for the empty assistant message logic below